packages/domain-picker/test/use-your-domain-item.tsx

```diff
@@ -7,29 +7,37 @@
  */
 import * as React from 'react';
 import { render, screen, fireEvent } from '@testing-library/react';
-import '@testing-library/jest-dom/extend-expect';
 
 /**
  * Internal dependencies
  */
 import UseYourDomainItem from '../src/domain-picker/use-your-domain-item';
 
 describe( '<UseYourDomainItem />', () => {
-	test( 'Component renders with correct messaging', () => {
+	it( 'should render with correct messaging', () => {
+		render( <UseYourDomainItem onClick={ jest.fn() } /> );
+
+		expect( screen.getByText( 'Already own a domain?' ) ).toBeInTheDocument();
+		expect( screen.getByText( "You can use it as your site's address." ) ).toBeInTheDocument();
+	} );
+
+	it( 'should fire onClick callback when the button is clicked', () => {
 		const onClick = jest.fn();
 
 		render( <UseYourDomainItem onClick={ onClick } /> );
 
-		expect( screen.getByText( 'Already own a domain?' ) ).toBeTruthy();
-		expect( screen.getByText( "You can use it as your site's address." ) ).toBeTruthy();
+		fireEvent.click( screen.getByText( 'Use a domain I own' ) );
+
+		expect( onClick ).toHaveBeenCalledTimes( 1 );
 	} );
 
-	test( 'Component fires onClick callback onClick', () => {
+	it( 'should fire onClick callback when the container is clicked', () => {
 		const onClick = jest.fn();
 
 		render( <UseYourDomainItem onClick={ onClick } /> );
 
-		fireEvent.click( screen.getByText( 'Use a domain I own' ) );
-		expect( onClick ).toHaveBeenCalled();
+		fireEvent.click( screen.getByTestId( 'use-domain-i-own-wrapper' ) );
```

Comment:
According to [Kent C. Dodds](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-the-wrong-query), a better alternative for `getByTestId` is `getByRole`. This query resembles the user's behavior much better and that's what we have to aim for in integration testing. 

We can query for the `ArrowButton` inside the `UseYourDomainItem` since that event bubbles to the container (see #51074). See https://testing-library.com/docs/queries/about/#priority for more information.

Reply:
We can't `getByRole` in this case, since the underlying element is a simple `div` without any role. I agree that the markup needs improving, but that should be treated as a separate issue.

In the meantime, I still went ahead and tried to use better alternatives in the tests:

- I was able to use `getByRole` when selecting buttons (https://github.com/Automattic/wp-calypso/pull/50889/commits/261711b74db3ce4d8a9cfb020393afa4fbc992ad)
- For the specific case highlighted by this comment, I switched from `getByTestId` to `getByText` — at least a way of testing that is closer to how the user interacts with the component (https://github.com/Automattic/wp-calypso/pull/50889/commits/617c7dc0bf34ad54de44c2c377ebc51cc5a5bf58)